libs/wingc/src/type_check.rs

```diff
@@ -3027,6 +3059,38 @@ impl<'a> TypeChecker<'a> {
 			Err(type_error) => self.variable_error(type_error),
 		}
 	}
+
+	fn resolve_user_defined_type(
+		&mut self,
+		user_defined_type: &UserDefinedType,
+		env: &SymbolEnv,
+		statement_idx: usize,
+	) -> Result<TypeRef, TypeError> {
+		// Attempt to resolve the type from the current environment
+		// otherwise, attempt to resolve the type from any jsii libs
+		let res = resolve_user_defined_type(user_defined_type, env, statement_idx);
+		if res.is_ok() {
+			return res;
+		}
+
+		for jsii in &*self.jsii_imports {
```

Comment:
If we have N jsii libraries imported in a Wing source file, would this `for` loop mean that we are doing an O(n) operation each time the compiler tries to resolve a user defined type? curious if we can avoid this, or provably bound the number of times this function is called in some way

Reply:
It's probably worth a benchmark. The alternative here would be to use a hashmap.
However, even though hashmaps have constant big O lookups, for small `n` the performance for plain old vectors may still be better. "small" here meaning like < 100. 
I don't think it'll be super common to have a large number of jsii imports.

Also, before this forloop it attempts to lookup the type in a O(1) way anyways. So this will not happen on every lookup.